src/spring-cloud/azext_spring_cloud/_params.py

```diff
@@ -74,8 +74,11 @@ def load_arguments(self, _):
         c.argument('name', name_type, help='Name of app.')
 
     with self.argument_context('spring-cloud app create') as c:
-        c.argument(
-            'is_public', arg_type=get_three_state_flag(), help='If true, assign public domain', default=False)
+        c.argument('assign_endpoint', arg_type=get_three_state_flag(),
+                   help='If true, assign endpoint URL for direct access.', default=False,
+                   options_list=['--assign-endpoint', c.deprecate(target='--is-public', redirect='--assign-endpoint', hide=True)])
+        c.argument('is_public', arg_type=get_three_state_flag(),
```

Comment:
Remove the argument definition for `is_public`, there is no argument `is_public` now, but argument `assign_endpoint` with a deprecated option `--is-public`.